iphone/Classes/GeolocationModule.m

```diff
@@ -869,7 +869,10 @@ - (void)requestLocationPermissioniOS7:(id)args {
 -(void)requestLocationPermissions:(id)args
 {
     if (![TiUtils isIOS8OrGreater]) {
-        [self requestLocationPermissioniOS7:args];
+        // It is required that delegate is created and permission is presented in main thread.
+        TiThreadPerformOnMainThread(^{
+            [self requestLocationPermissioniOS7:args];
+        }, NO);
```

Comment:
Alright. But if thats the case, we definitely need to make the Xcode builds behave the same as the actual `ti build -p ios` builds. Would it make sense to define it by default in the Xcode project? Should be the same with `<use-autolayout/>` that triggers a different behavior.